data_steward/tools/delete_stale_test_datasets.py

```diff
@@ -0,0 +1,103 @@
+"""
+Delete stale datasets in test environment. 
+"""
+from datetime import datetime, timezone
+import logging
+import os
+
+# Project imports
+from utils import bq, pipeline_logging
+
+LOGGER = logging.getLogger(__name__)
+
+
+def _check_project(bq_client):
+    """
+    Check if the project is set to test.
+
+    :param bq_client: Client
+    :return: None if the project is set properly
+    :raise: ValueError if project is not 'aou-res-curation-test'
+    """
+
+    if 'test' not in bq_client.project:
+        raise ValueError(
+            f'Wrong project: {bq_client.project}. This script runs only for test environment'
+        )
+
+    return None
+
+
+def _filter_stale_datasets(bq_client, first_n: int = None):
+    """
+    Get the first n datasets that meet all of the following criteria:
+    1. Datasets older than 90 days
+    2. Empty datasets
+
+    :param bq_client: Client
+    :param first_n: number of datasets to return. If not specified, return all.
+    :return: list of dataset names that are stale
+    """
+
+    stale_datasets, n = [], 0
+    now = datetime.now(timezone.utc)
+
+    dataset_names = [
+        dataset.dataset_id for dataset in bq_client.list_datasets()
+    ]
+
+    for dataset_name in dataset_names:
+
+        if first_n and n >= first_n:
+            break
+
+        dataset_created = bq_client.get_dataset(dataset_name).created
+
+        if (now - dataset_created).days <= 90:
+            continue
+
+        LOGGER.critical(
+            f"{dataset_name}, {type(bq_client.list_tables(dataset_name))}")
+        LOGGER.critical(
+            f"{dataset_name}, {dir(bq_client.list_tables(dataset_name))}")
+        LOGGER.critical(
+            f"{dataset_name}, {bq_client.list_tables(dataset_name).pages}")
+        LOGGER.critical(
+            f"{dataset_name}, {dir(bq_client.list_tables(dataset_name).item_to_value)}"
+        )
+        #if next(bq_client.list_tables(dataset_name), None):
+        #    continue
+        if list(bq_client.list_tables(dataset_name)) == 0:
+            continue
+
+        stale_datasets.append(dataset_name)
+        LOGGER.info(
+            f"{n}: stale_dataset={dataset_name}, time_created={dataset_created}."
+        )
+        n += 1
+
+    return stale_datasets
+
+
+def main():  # change to run-deletion or something
+
+    pipeline_logging.configure(logging.INFO, add_console_handler=True)
+
+    bq_client = bq.get_client(os.environ.get('GOOGLE_CLOUD_PROJECT'))
+
+    _check_project(bq_client)
+
+    stale_datasets = _filter_stale_datasets(bq_client, first_n=100)
+
+    for stale_dataset in stale_datasets:
+        LOGGER.info(
```

Comment:
`INFO` is enough for creating output to the console.

![image](https://user-images.githubusercontent.com/90213198/145261312-0712b471-6dba-4e59-9cc3-0a1e01fc836d.png)

https://app.circleci.com/pipelines/github/all-of-us/curation/4064/workflows/60fe0c49-37ad-4e39-a3c4-d4154e877f2c/jobs/12135